From ee01c944e997bd3fcd4c9796a83647bdffe0dd12 Mon Sep 17 00:00:00 2001
From: Ramya Gnanasekar <quic_rgnanase@quicinc.com>
Date: Tue, 14 Mar 2023 14:23:31 +0530
Subject: [PATCH] mac80211: update radar_required in channel context after
 channel switch

When the channel switch has been made, a vif will use the
channel context which was reserved. Channel context and channel
definition in sdata will also be updated accordingly.
In this case, radar required parameter passed from NL stored as
reserved_radar_required should also be updated to sdata.

When channel is switched from 80 MHz to 160 MHz where primary 80 MHz
is non DFS and secondary 80 MHz channels are DFS channels, radar
required in channel switch parameters will be true but it is not
updated to sdata when using reserved channel context.

This flag is used by ath12k driver to inform that the channel is radar
enabled to firmware during vdev start/restart.
Update the radar required flag to fix the same.

Signed-off-by: Ramya Gnanasekar <quic_rgnanase@quicinc.com>

diff --git a/net/mac80211/chan.c b/net/mac80211/chan.c
index 7ed61d2..7b03b27 100644
--- a/net/mac80211/chan.c
+++ b/net/mac80211/chan.c
@@ -1169,6 +1169,7 @@ ieee80211_vif_use_reserved_reassign(struct ieee80211_sub_if_data *sdata)
 
 	list_del(&sdata->reserved_chanctx_list);
 	sdata->reserved_chanctx = NULL;
+	sdata->radar_required = sdata->reserved_radar_required;
 
 	err = drv_switch_vif_chanctx(local, vif_chsw, 1,
 				     CHANCTX_SWMODE_REASSIGN_VIF);
-- 
2.17.1

